### PR TITLE
Repair BigInt.toDouble()

### PR DESCRIPTION
**Pull request by ilikerobots**
Fixes for #14

### CODE REVIEW

File: lib/bigint.dart [hunk 1]
```diff
@@ -103,10 +103,11 @@ class BigInt implements Comparable<BigInt> {
   }
 
   double toDouble() {
-    double result = 1.0;
+    double result = 0.0;
     for (int i = 0; i < value.length; i++) {
       result += value[i] * pow(_BASE, i);
     }
+    if (isNegative) result = -1.0 * result;
```
Comment (a14n) [inline lib/bigint.dart]: `result = -result;` should be more efficient.

File: test/bigint_test.dart [hunk 2]
```diff
@@ -31,6 +31,16 @@ main() {
...
+  test('toDouble()', () {
+    [ '0', '1', '-1', '2', '-2', '23', '-23', '-0',
+      '3187801890382', '-3187801890382',
+    ].forEach((n) =>
+        expect(p(n).toDouble().toString().replaceAll(".0", ""), n));
+    expect(p('31878018903828899277492024491376690701584023926880').toDouble().toString(),
+        allOf([startsWith('3.187801890382'), endsWith('e+49')]));
+    expect(p('-31878018903828899277492024491376690701584023926880').toDouble().toString(),
+        allOf([startsWith('-3.187801890382'), endsWith('e+49')]));
+  });
```
Comment (a14n) [inline test/bigint_test.dart]: Can you use `double.parse` instead of String comparaison ?
Something like this : 

```dart
  test('toDouble()', () {
    [
      '0',
      '1',
      '2',
      '23',
      '3187801890382',
      '31878018903828899277492024491376690701584023926880',
    ]
        .expand((e) => [e, '-$e']) // also test negative values
        .forEach((n) => expect(p(n).toDouble(), equals(double.parse(n))));
  });
```

**Comment by a14n** [comment]
I will address my comments directly.

Thanks again for catching this issue and provide a fix :)

File: test/bigint_test.dart (hunk 2 above)
Comment (a14n) [inline test/bigint_test.dart]: Ok I haven't test in browser and I understand this String check now :)

**Comment by ilikerobots** [comment]
Sorry for the delay, I was just about to address your comments.  But thaks for the quick update!
